### PR TITLE
spring

### PR DESCRIPTION
**Pull request by soaptherope**

### CODE REVIEW

File: src/main/java/org/andersen/hotel/controller/ApartmentController.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    @PutMapping("/{id}/reserve")
+    public ResponseEntity<Void> reserveApartment(@PathVariable Long id, @RequestParam String nameOfClient) {
+        try {
+            apartmentService.reserveApartment(id, nameOfClient);
+            return ResponseEntity.ok().build();
+        } catch (ApartmentWithIdNotFoundException e) {
+            return ResponseEntity.notFound().build();
+        } catch (IncorrectStatusOfApartmentException e) {
+            return ResponseEntity.badRequest().build();
+        }
```
Comment (Alexmansar) [inline src/main/java/org/andersen/hotel/controller/ApartmentController.java]: in general this try catch logic should be in service layer

File: src/main/java/org/andersen/hotel/service/ApartmentService.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package org.andersen.hotel.service;
+
+import org.andersen.hotel.model.Apartment;
+import org.springframework.data.domain.Sort;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface ApartmentService {
```
Comment (Alexmansar) [inline src/main/java/org/andersen/hotel/service/ApartmentService.java]: in general, you can create bease service with generic type

File: src/main/java/org/andersen/hotel/controller/ApartmentController.java (hunk 1 above)
Comment (Alexmansar) [reply src/main/java/org/andersen/hotel/controller/ApartmentController.java]: or, https://www.baeldung.com/exception-handling-for-rest-with-spring
this silution will be better

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@soaptherope Hi))In general good job)) I have some comments))
about tests - if you want to use this repo like a pet project, please, add it) not for now, but in general

File: src/main/java/org/andersen/hotel/service/ApartmentService.java (hunk 2 above)
Comment (soaptherope) [reply src/main/java/org/andersen/hotel/service/ApartmentService.java]: in that case, what do i do with specific methods such as reserve apartment? have 2 interfaces (one generic and one specific) or just 1 generic and write specific methods in impl class?

Comment (soaptherope) [reply src/main/java/org/andersen/hotel/service/ApartmentService.java]: got it, delegate interface, didnt know we can extend them))

Comment (Alexmansar) [reply src/main/java/org/andersen/hotel/service/ApartmentService.java]: 1 base) after that new one with additional methods
and just extend it))

**Comment by soaptherope** [comment]
i removed custom exceptiosn for simplicity to focus on exception handler

**Comment by soaptherope** [comment]
also for updating price and name, i could theoretically parse integer from String to make it generic too but maybe not a good idea

**Review by Alexmansar — APPROVED** [review]
@soaptherope Good job)) does it work?
I mean without Transactions))

**Comment by soaptherope** [comment]
you are right, forgot that we modify when reserving or releasing and updating afterwards. are there any other methods i should mark as transactional?

**Comment by soaptherope** [comment]
perhaps for updating price and name, but its a single change on one field

**Comment by soaptherope** [comment]
{E1C7D3F2-3DD6-4E5A-ADF3-30E2E4E09842}
it works otherwise

**Comment by Alexmansar** [comment]
@soaptherope in general we use transaction on the class level for all public methods

**Comment by soaptherope** [comment]
for crudservice and apartmentservice it makes sense, but added to hotelservice just in case) why not, jpa makes basic operations transcational anyways

**Review by Alexmansar — APPROVED** [review]

**Pull request merged**